sharing_portal/views.py

```diff
@@ -558,4 +863,39 @@ def _request_artifact_dois(artifact, request_forms=[]):
             return True
         return False
     except Exception:
-        LOG.exception('Failed to request DOI for artifact {}'.format(artifact.pk))
+        LOG.exception("Failed to request DOI for artifact {}".format(artifact.pk))
+
+
+def create_supplemental_project(request, artifact):
+    mapper = ProjectAllocationMapper(request)
+
+    pi = artifact.project.pi
+    supplemental_project = {
+        "nickname": f"reproducing_{artifact.id}",
+        "title": f"Reproducing '{artifact.title}'",
+        "description": f"This project is for reproducing the artifact '{artifact.title}'",
+        "typeId": artifact.project.type.id,
+        "fieldId": artifact.project.field.id,
+        "piId": artifact.project.pi.id,
+    }
+    # Approval code is commented out during initial preview release.
+    allocation_data = {
+        "resourceId": 39,
```

Comment:
hmm?

Reply:
This is copied from how we do it when creating a project.

Reply:
Ah right, this is a weird legacy TAS (TACC Accounting System) thing, I think.